### ISSUE TITLE
CMake 3.20.0 triggers warnings concerning policies CMP0115 and CMP0116

### ISSUE THREAD

**Issue by fabianschuiki** [llvm/circt#849]
I just updated to CMake 3.20.0 on my machine and now run into the following warnings:

- CMP0115
  ```
  CMake Warning (dev) at tools/circt-rtl-sim/CMakeLists.txt:16 (add_custom_target):
    Policy CMP0115 is not set: Source file extensions must be explicit.  Run
    "cmake --help-policy CMP0115" for policy details.  Use the cmake_policy
    command to set the policy and suppress this warning.
  
    File:
  
      [...]/circt/tools/circt-rtl-sim/circt-rtl-sim.py.in
  This warning is for project developers.  Use -Wno-dev to suppress it.
  ```
- CMP0116
  ```
  CMake Warning (dev) at llvm/llvm/cmake/modules/TableGen.cmake:94 (add_custom_command):
    Policy CMP0116 is not set: Ninja generators transform DEPFILEs from
    add_custom_command().  Run "cmake --help-policy CMP0116" for policy
    details.  Use the cmake_policy command to set the policy and suppress this
    warning.
  Call Stack (most recent call first):
    llvm/build/lib/cmake/mlir/AddMLIR.cmake:2 (tablegen)
    llvm/build/lib/cmake/mlir/AddMLIR.cmake:13 (mlir_tablegen)
    cmake/modules/AddCIRCT.cmake:4 (add_mlir_dialect)
    include/circt/Dialect/StaticLogic/CMakeLists.txt:1 (add_circt_dialect)
  This warning is for project developers.  Use -Wno-dev to suppress it.
  ```

The [changelog](https://cmake.org/cmake/help/latest/release/3.20.html#other-changes) for CMake 3.20 reports:
> - Source file extensions must now be explicit. See policy CMP0115 for details.
> - Ninja generators now transform the DEPFILE generated by an add_custom_command(). See policy CMP0116 for details.

Details on the policies:
- [CMP0115](https://cmake.org/cmake/help/latest/policy/CMP0115.html)
- [CMP0116](https://cmake.org/cmake/help/latest/policy/CMP0116.html)
